packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -352,22 +388,47 @@ export class Canary extends cdk.Resource {
     };
   }
 
+  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
+    if (!props.environmentVariables) {
+      return undefined;
+    }
+    return {
+      environmentVariables: props.environmentVariables,
+    };
+  }
+
   /**
-   * Returns a canary schedule object
-   */
+  * Returns a canary schedule object
+  */
   private createSchedule(props: CanaryProps): CfnCanary.ScheduleProperty {
     return {
       durationInSeconds: String(`${props.timeToLive?.toSeconds() ?? 0}`),
       expression: props.schedule?.expressionString ?? 'rate(5 minutes)',
     };
   }
 
-  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+  private createVpcConfig(props: CanaryProps): CfnCanary.VPCConfigProperty | undefined {
+    if (!props.vpc) {
       return undefined;
     }
+
+    const { subnetIds } = props.vpc.selectSubnets(props.vpcSubnets);
+    let securityGroups: ec2.ISecurityGroup[];
+
+    if (props.securityGroups && props.securityGroups.length > 0) {
+      securityGroups = props.securityGroups;
```

Comment:
What happens if the user provides `props.securityGroups = []`?  What do we want to happen?

Reply:
The short answer is I don't know, but if I had to hazard a guess it would attach to the default security group for a given VPC which might cause surprising behavior. Though I guess if we documented it and explicitly said `[]` == `default sg` that wouldn't be too surprising?

Reply:
To clarify what happens presently with this PR, we fallback to creating an sg which allows all egress traffic.

Reply:
Actually, I missed that you have `props.securityGroup.length > 0`. I think that is fine.